app/views/entities/show.rb

```diff
@@ -38,24 +39,34 @@ def include_nav_searchbar
       true
     end
 
-    # TODO
-    # def head_meta
-    #   mustache[:head_meta] ||= begin
-    #     title = page_title
-    #     head_meta = [
-    #       { meta_name: 'description', content: head_meta_description },
-    #       { meta_property: 'fb:appid', content: '185778248173748' },
-    #       { meta_name: 'twitter:card', content: 'summary' },
-    #       { meta_name: 'twitter:site', content: '@EuropeanaEU' },
-    #       { meta_property: 'og:sitename', content: title },
-    #       { meta_property: 'og:description', content: head_meta_description },
-    #       { meta_property: 'og:url', content: collection_url(@collection.key) }
-    #     ]
-    #     head_meta << { meta_property: 'og:title', content: title } unless title.nil?
-    #     head_meta << { meta_property: 'og:image', content: @landing_page.og_image } unless @landing_page.og_image.blank?
-    #     head_meta + super
-    #   end
-    # end
+    def head_meta
+      mustache[:head_meta] ||= begin
+        title = page_title
+        head_meta = [
+          { meta_name: 'description', content: head_meta_description },
+          { meta_property: 'fb:appid', content: '185778248173748' },
+          { meta_name: 'twitter:card', content: 'summary' },
+          { meta_name: 'twitter:site', content: '@EuropeanaEU' },
+          { meta_property: 'og:sitename', content: title },
+          { meta_property: 'og:description', content: head_meta_description },
+          { meta_property: 'og:url', content: request.original_url }
+        ]
+        head_meta << { meta_property: 'og:title', content: title } unless title.nil?
+        head_meta << { meta_property: 'og:image', content: og_image } unless og_image.nil?
+        head_meta + super
+      end
+    end
+
+    def head_meta_description
+      mustache[:head_meta_description] ||= begin
+        truncate(entity_description, length: 350, separator: ' ')
+      end
+    end
+
+    def og_image
+      thumbnail = entity_thumbnail
```

Comment:
actually because this is being called twice, once to check if it's nil, and once to assign it into the content of the meta tag, this should be wrapped in a block too.
```
mustache[:og_image] ||= begin
  ...
end
```